test/extensions/filters/common/expr/context_test.cc

```diff
@@ -357,6 +357,31 @@ TEST(Context, ResponseAttributes) {
   }
 }
 
+TEST(Context, ConnectionFallbackAttributes) {
+  NiceMock<StreamInfo::MockStreamInfo> info;
+  ConnectionWrapper connection(info);
+  UpstreamWrapper upstream(info);
+  {
+    auto value = connection[CelValue::CreateStringView(Undefined)];
+    EXPECT_FALSE(value.has_value());
+  }
+
+  {
+    auto value = connection[CelValue::CreateStringView(ID)];
+    EXPECT_FALSE(value.has_value());
+  }
+
+  {
+    auto value = upstream[CelValue::CreateStringView(Undefined)];
+    EXPECT_FALSE(value.has_value());
```

Comment:
why is this repeated?

Reply:
It uses `upstream` instead of `connection` here.